File: 01/solution.py
with open('input.txt') as fp:
    data = list(map(lambda x: sum(int(y) for y in x.split('\n')), fp.read().strip().split('\n\n')))


# Part I
print(max(data))

# Part II
print(sum(sorted(data, reverse=True)[:3]))

File: 01/Solution.hs
import Data.List

reader x (a, b) | x == "" = (a ++ [b], 0)
                | otherwise = (a, b + (read x))

parse raw = reverse $ (fst i) ++ [snd i]
    where i = foldr reader ([], 0) (lines raw)

solve = ((\x -> (part1 x, part2 x)) . parse) <$> readFile "input.txt"
    where part1 = head . reverse . sort
    where part2 = sum . (take 3) . reverse . sort
